lib/segment/src/types.rs

```diff
@@ -2010,6 +2041,18 @@ impl Filter {
         };
         Filter {
             should: merge_component(self.should.clone(), other.should.clone()),
+            // TODO
+            min_should: {
+                match (self.min_should.clone(), other.min_should.clone()) {
+                    (None, None) => None,
+                    (Some(this), None) => Some(this),
+                    (None, Some(other)) => Some(other),
+                    (Some(mut this), Some(other)) => {
+                        this.conditions.extend(other.conditions);
+                        Some(this) // min_count follows self.min_should.min_count
+                    }
+                }
+            },
```

Comment:
I think lack understanding of merging Filters. Is there any use case when merging min_should clauses is required? The only cases that I found in source code was merging `must` or `must_not` filters.